PWGCF/Tasks/dptdptcorrelations-simchl.cxx

```diff
@@ -1470,40 +1470,188 @@ struct TracksAndEventClassificationQAGen {
   }
 };
 
+/// \brief Checks the correspondence generator level <=> detector level
+struct CheckGeneratorLevelVsDetectorLevel {
+  OutputObj<TH1F> multirecDeltaEta{TH1F("mrDeltaEta", "#Delta#eta multirec tracks;#Delta#eta", 100, -2, 2)};
+  OutputObj<TH1F> multirecDeltaPhi{TH1F("mrDeltaPhi", "#Delta#varphi multirec tracks;#Delta#varphi", 100, 0, 2.0f * static_cast<float>(M_PI))};
+  OutputObj<TH1F> multirecDeltaPt{TH1F("mrDeltaPt", "#Delta#it{p}_{T} multirec tracks;#Delta#it{p}_{T}", 100, 0, 4)};
+  OutputObj<TH1F> multirecDistribution{TH1F("multirec", "Multiple reconstruction; ##/particle; counts", 11, -0.5f, 10.5f)};
+  OutputObj<TH1F> multirecDeltaEta_nc{TH1F("mrDeltaEta_nc", "#Delta#eta multirec tracks;#Delta#eta", 100, -2, 2)};
+  OutputObj<TH1F> multirecDeltaPhi_nc{TH1F("mrDeltaPhi_nc", "#Delta#varphi multirec tracks;#Delta#varphi", 100, 0, 2.0f * static_cast<float>(M_PI))};
+  OutputObj<TH1F> multirecDeltaPt_nc{TH1F("mrDeltaPt_nc", "#Delta#it{p}_{T} multirec tracks;#Delta#it{p}_{T}", 100, 0, 4)};
+  OutputObj<TH1F> multirecDistribution_nc{TH1F("multirec_nc", "Multiple reconstruction; ##/particle; counts", 11, -0.5f, 10.5f)};
+  TDatabasePDG* fPDG;
+
+  void init(InitContext const& context)
+  {
+    fPDG = TDatabasePDG::Instance();
+  }
+
+  void process(soa::Join<aod::Tracks, aod::McTrackLabels> const& tracks, aod::McParticles const& mcParticles)
+  {
+    constexpr float TWOPI = 2.0F * static_cast<float>(M_PI);
+
+    std::vector<std::vector<int64_t>> mclabelpos(mcParticles.size());
+    std::vector<std::vector<int64_t>> mclabelneg(mcParticles.size());
+    std::vector<std::vector<int64_t>> mclabelpos_negcoll(mcParticles.size());
+    std::vector<std::vector<int64_t>> mclabelneg_negcoll(mcParticles.size());
+
+    size_t nreco = tracks.size();
+    size_t ngen = 0;
+
+    for (auto& part : mcParticles) {
+      auto pdgpart = fPDG->GetParticle(part.pdgCode());
+      if (pdgpart != nullptr) {
+        float charge = (pdgpart->Charge() / 3 >= 1) ? 1.0 : ((pdgpart->Charge() / 3 <= -1) ? -1.0 : 0.0);
+        if (charge != 0.0) {
+          ngen++;
+        }
+      }
+    }
+
+    // Let's go through the reco-gen mapping to detect multi-reconstructed particles
+    // For the time being we are only interested in the information based on the reconstructed tracks
+    LOGF(info, "New chunk with %d generated charged particles and %d reconstructed tracks", ngen, nreco);
+
+    for (auto& track : tracks) {
+      int64_t recix = track.globalIndex();
+      int32_t label = track.mcParticleId();
+
+      // LOGF(info, "Track with global Id %d and collision Id %d has label %d associated to MC collision %d", recix, track.collisionId(), label, track.mcParticle().mcCollisionId());
+      if (track.collisionId() < 0) {
```

Comment:
This can be negative for unassigned tracks. In this case the collisionId() will appear many times. 

Say the simulation has produced 10 collisions, then all tracks will point to their respective collision (id 0 ... 9)
All unassigned tracks will have -1

If we merge several files then the entries of the first file will have -1, the second file -2 and so on

Reply:
For the tracks with negative `collisionId()` we only keep track of the positive/negative labels but so far, unless it were needed, we have not planned to check anything additionally

Reply:
Actually, after reviewing it again, we also keep track of the multi-reconstruction when the `collisionId()` is negative but probably this will not be valid information

Reply:
Thanks for clarifying the different negative collision Ids. I was a bit surprised when found them in the pp nightly
There were only one file but I guess there were several DFs